spec/gollum_git_repo_modification_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+require 'rspec/expectations'
+
+RSpec::Matchers.define :a_blob_named do |expected|
+  match do |actual|
+    actual.is_a?(Gollum::Git::Blob) && actual.name == expected
+  end
+end
+
+describe Gollum::Git::Repo do
+
+  after(:each) do
+    if parent_commit != repo.head.commit
+      repo.update_ref(repo.head.name, parent_commit.id)
+    end
+  end
+
+  let(:repo) { Gollum::Git::Repo.new(fixture('dot_bare_git'), :is_bare => true) }
+
+  let(:index) do
+    repo.index.read_tree(repo.head.commit.tree.id)
+    repo.index
+  end
+
+  let(:actor) { Gollum::Git::Actor.new("Tom Werner", "tom@example.com") }
+
+  let(:message) { 'Test commit' }
+
+  let(:parent_commit) { repo.head.commit }
+
+  def commit_index
+    index.commit(message, [parent_commit], actor, nil, repo.head.name)
+  end
+
+  describe "#head#commit" do
+
+    context "after committing" do
+
+      before(:each) do
+        index.add('Add.txt', 'Some data')
+        commit_index
```

Comment:
The pollution issue needs to be dealt with, and I think your proposal is a good strategy for doing so. An alternative would be to include these more substantial tests in `gollum-lib`'s test suite, as there we already have the mechanisms in place for setting up, committing to and cleaning up temporary repositories. The adapter specs were originally designed as the poor man's way of defining an interface. But I also see the appeal of having these tests in the `adapter_specs` (especially since they're written in rspec). What are your thoughts on this, @dometto ?